uploaders/common.go

```diff
@@ -0,0 +1,322 @@
+	log.Infof("Register %s on Bitrise...", profile.Name)
+
+	requestURL := createRequestURLForProvProfSlug(appSlug)
+
+	fields := map[string]interface{}{
+		"upload_file_name": profile.Name,
+		"upload_file_size": provisioningProfSize,
+	}
+
+	headers := map[string]string{
+		"Authorization": "token " + accessToken,
+	}
+
+	request, err := createRequest(requestURL, fields, Post, headers)
+	if err != nil {
+		failWithMessage("Failed to create request, error: %#v", err)
+	}
+
+	// Response struct
+	requestResponse := RegisterProvisioningProfileResponse{}
+	responseStatusCode := -1
+
+	//
+	// Perform request
+	if err := retry.Times(1).Wait(5 * time.Second).Try(func(attempt uint) error {
+		body, statusCode, err := performRequest(request)
+		if err != nil {
+			log.Warnf("Attempt (%d) failed, error: %s", attempt+1, err)
+			if !strings.Contains(err.Error(), "failed to perform request") {
+				log.Warnf("Response status: %d", statusCode)
+				log.Warnf("Body: %s", string(body))
+			}
+			return err
+		}
+
+		responseStatusCode = statusCode
+
+		// Parse JSON body
+		if err := json.Unmarshal([]byte(body), &requestResponse); err != nil {
+			return fmt.Errorf("failed to unmarshal response (%s), error: %s", body, err)
+		}
+		return nil
+
+	}); err != nil {
+		log.Errorf("Fetching list failed %s", err)
+		return RegisterProvisioningProfileResponseData{}, err
+	}
+
+	// Success
+	log.Donef("Request succeeded with status code: %d", responseStatusCode)
+	logDebugPretty(requestResponse)
+
+	return requestResponse.Data, nil
+}
+
+// UploadProvisioningProfile ...
+func UploadProvisioningProfile(responseData RegisterProvisioningProfileResponseData, outputDirPath string, exportFileName string) error {
+	log.Infof("Upload %s to Bitrise...", exportFileName)
+
+	requestURL := responseData.UploadURL
+
+	files := map[string]string{
+		responseData.UploadFileName: (outputDirPath + "/" + exportFileName),
+	}
+
+	request := createUploadRequest(requestURL, files, Put, map[string]string{})
+
+	// Response struct
+	responseStatusCode := -1
+
+	//
+	// Perform request
+	if err := retry.Times(1).Wait(5 * time.Second).Try(func(attempt uint) error {
+		body, statusCode, err := performRequest(request)
+		if err != nil {
+			log.Warnf("Attempt (%d) failed, error: %s", attempt+1, err)
+			if !strings.Contains(err.Error(), "failed to perform request") {
+				log.Warnf("Response status: %d", statusCode)
+				log.Warnf("Body: %s", string(body))
+			}
+			return err
+		}
+
+		responseStatusCode = statusCode
+
+		return nil
+
+	}); err != nil {
+		return err
+	}
+
+	// Success
+	log.Donef("Request succeeded with status code: %d", responseStatusCode)
+
+	return nil
+}
+
+func createUploadRequest(url string, files map[string]string, requestMethod httpMethod, headers map[string]string) *http.Request {
```

Comment:
if there are several 'commonly' used functions you can move them into: common.go file

Reply:
Are you sure about this?
I think it would be better if every "network type" function will be under the uploaders package and not the cmd package.